services/bonita-page/bonita-page-impl/src/main/java/org/bonitasoft/engine/page/impl/PageMappingServiceImpl.java

```diff
@@ -118,7 +132,15 @@ public SPageMapping get(String key) throws SObjectNotFoundException, SBonitaRead
     }
 
     @Override
-    public SPageURL resolvePageURL(SPageMapping pageMapping, Map<String, Serializable> context) throws SExecutionException {
+    public SPageURL resolvePageURL(SPageMapping pageMapping, Map<String, Serializable> context) throws SExecutionException, SAuthorizationException {
+        final List<String> pageAuthorizationRules = pageMapping.getPageAuthorizationRules();
+        final List<AuthorizationRule> rules = getRules(pageAuthorizationRules);
```

Comment:
Maybe we can avoid to create a a list of rule and only iterate on the pageAuthorizationRules